content/livesync/frontend.textile

```diff
@@ -0,0 +1,373 @@
+
+Instantiate a model using a unique channel name. A unique channel name identifies the model on the client and corresponds to the "channel":/channels name used to subscribe to state updates from the backend:
+
+```[javascript]
+const model = modelsClient.models.get({
+  name: 'Posts',
+  channelName: 'posts',
+  sync,
+  merge,
+});
+```
+
+TODO: Diagram of how the model components work together
+
+h2(#designing). Designing your Channels and Models
+
+In general, when designing your channels and models, aim for a 1:1:1 relationship between channels, models, and domain objects. Channels facilitate data partitioning and access control. For instance, in a task management app, each task can be represented by a model with its own channel, ensuring updates are specific to each task. Ably's "Token Capabilities":/auth/capabilities allow fine-grained access control.
+
+Having each model on its own channel reduces processing overhead, ensuring models only receive relevant messages. While multiple models can share a channel, it requires careful message filtering. Channels aid scalability, but finding the right balance between channels, models, and message frequency is crucial.
+
+For infrequently updated models, grouping them on the same channel may be viable, while frequently changing models benefit from individual channels. Starting with one channel per model and one model per domain object is a good baseline, adjusting as needed.
+
+h2(#create-sync). Create a Sync function
+
+The sync function instructs your model to initialize with the latest data from the backend and is necessary when creating a model. It can be any function that optionally accepts parameters and returns a promise with the latest state of your data model, along with a "@sequenceId@":#sequence-id.
+
+The following is an example sync function that fetches the latest model state from your database by calling `/api/posts` endpoint in your backend:
+
+```[javascript]
+async function sync() {
+  const result = await fetch(`/api/posts`);
+
+  return result.json();
+}
+```
+
+The following example is an example output from the sync function:
+
+```[json]
+{
+  "sequenceId": "1",
+  "data": [
+    {
+      "id": 123,
+      "text": "Hello World",
+      "comments": []
+    },
+    {
+      "id": 124,
+      "text": "Another post",
+      "comments": []
+    }
+  ]
+}
+```
+
+The sync function must be registered when a model is "instantiated:":#create
+
+```[javascript]
+const model = modelsClient.models.get({
+  name: 'Posts',
+  channelName: 'posts',
+  sync,
+  merge,
+});
+```
+
+Call the sync function directly on the model to bootstrap its state during the initial page load:
+
+```[javascript]
+await model.sync();
+```
+
+The SDK automatically attempts to re-execute the sync function in the case of an error, as specified by the @syncOptions.retryStrategy@ defined in your "@ClientOptions@":/livesync/models#options.
+
+h3(#sequence-id). Sequence ID
```

Comment:
What is the thinking behind placing this section at this point in the page, OOI?

Reply:
I thought adding the sequence ID under the sync section was important because you cannot have the `sync` function without the sequence ID, If you only brief over the sequence ID here, it'll leave the developer somewhat confused over what it is. 

Is there an issue with this or do you think it should be elsewhere?

Reply:
Ok. That makes sense. I've added a sentence that explains what where the sequence id comes from at the beginning.